eisbuk-admin/functions/src/https.ts

```diff
@@ -18,83 +18,71 @@ import {
 import { checkUser, createSMSReqOptions } from "./utils";
 
 /**
- * Stores email data to `emailQueue` collection, triggering firestore-send-email extension.
+ * An object of key-value (string-string) pairs
+ * where the key is a required field and value is an error message to
+ * display if field not present
  */
-export const sendEmail = functions
-  .region("europe-west6")
-  .https.onCall(async (payload: Partial<SendMailPayload>, { auth }) => {
-    // check payload
-    if (!payload) {
-      throw new functions.https.HttpsError(
-        "invalid-argument",
-        HTTPErrors.NoPayload
-      );
-    }
-
-    const { organization, ...email } = payload;
-
-    await checkUser(organization, auth);
+interface RequiredFieldsValidation {
+  [field: string]: string;
+}
 
-    // check payload
-    if (!email.to) {
-      throw new functions.https.HttpsError(
-        "invalid-argument",
-        SendEmailErrors.NoRecipient
-      );
-    }
-    if (!email.message?.html) {
-      throw new functions.https.HttpsError(
-        "invalid-argument",
-        SendEmailErrors.NoMsgBody
-      );
-    }
-    if (!email.message?.subject) {
+const checkRequiredFields = (
+  payload: Record<string, any> | undefined | null,
+  requiredFields: RequiredFieldsValidation
+) => {
+  if (!payload) {
+    throw new functions.https.HttpsError(
+      "invalid-argument",
+      HTTPErrors.NoPayload
+    );
+  }
+
+  Object.keys(requiredFields).forEach((field) => {
+    if (!payload[field]) {
       throw new functions.https.HttpsError(
         "invalid-argument",
-        SendEmailErrors.NoSubject
+        requiredFields[field]
       );
     }
+  });
+};
 
-    // add email to firestore, firing data trigger
-    await admin.firestore().collection(Collection.EmailQueue).doc().set(email);
+/**
+ * Stores email data to `emailQueue` collection, triggering firestore-send-email extension.
+ */
+export const sendEmail = functions
+  .region("europe-west6")
+  .https.onCall(
+    async ({ organization, ...email }: SendMailPayload, { auth }) => {
+      await checkUser(organization, auth);
+
+      checkRequiredFields(email, {
+        to: SendEmailErrors.NoRecipient,
```

Comment:
I propose we use a single `MissingParameter` exception, and when we instantiate and throw it, we specify the list of fields that were missing.